sentry_sdk/integrations/opentelemetry/span_processor.py

```diff
@@ -296,65 +310,76 @@ def _update_span_with_otel_data(self, sentry_span, otel_span):
         This should eventually happen on the server when ingesting the spans.
         """
-        for key, val in otel_span.attributes.items():
-            sentry_span.set_data(key, val)
-
         sentry_span.set_data("otel.kind", otel_span.kind)
 
         op = otel_span.name
         description = otel_span.name
 
-        http_method = otel_span.attributes.get(SpanAttributes.HTTP_METHOD, None)
-        db_query = otel_span.attributes.get(SpanAttributes.DB_SYSTEM, None)
-
-        if http_method:
-            op = "http"
-
-            if otel_span.kind == SpanKind.SERVER:
-                op += ".server"
-            elif otel_span.kind == SpanKind.CLIENT:
-                op += ".client"
-
-            description = http_method
-
-            peer_name = otel_span.attributes.get(SpanAttributes.NET_PEER_NAME, None)
-            if peer_name:
-                description += " {}".format(peer_name)
-
-            target = otel_span.attributes.get(SpanAttributes.HTTP_TARGET, None)
-            if target:
-                description += " {}".format(target)
-
-            if not peer_name and not target:
-                url = otel_span.attributes.get(SpanAttributes.HTTP_URL, None)
-                if url:
-                    parsed_url = urlparse(url)
-                    url = "{}://{}{}".format(
-                        parsed_url.scheme, parsed_url.netloc, parsed_url.path
-                    )
-                    description += " {}".format(url)
-
-            status_code = otel_span.attributes.get(
-                SpanAttributes.HTTP_STATUS_CODE, None
-            )
-            if status_code:
-                sentry_span.set_http_status(status_code)
-
-        elif db_query:
-            op = "db"
-            statement = otel_span.attributes.get(SpanAttributes.DB_STATEMENT, None)
-            if statement:
-                description = statement
+        if otel_span.attributes is not None:
+            for key, val in otel_span.attributes.items():
+                sentry_span.set_data(key, val)
+
+            http_method = otel_span.attributes.get(SpanAttributes.HTTP_METHOD)
+            http_method = cast("Optional[str]", http_method)
+
+            db_query = otel_span.attributes.get(SpanAttributes.DB_SYSTEM)
+            db_query = cast("Optional[str]", db_query)
+
+            if http_method:
+                op = "http"
+
+                if otel_span.kind == SpanKind.SERVER:
+                    op += ".server"
+                elif otel_span.kind == SpanKind.CLIENT:
+                    op += ".client"
+
+                description = http_method
+
+                peer_name = otel_span.attributes.get(SpanAttributes.NET_PEER_NAME, None)
+                if peer_name:
+                    description += " {}".format(peer_name)
+
+                target = otel_span.attributes.get(SpanAttributes.HTTP_TARGET, None)
+                if target:
+                    description += " {}".format(target)
```

Comment:
Because mypy doesn't complain 🙃 

Prolly because `target` is only used in the format string and whatever type it is, it'll have a `__str__` defined, so no type issues there. Whereas for example for `http_method` above, it's `__add__`ed to a string which is only supported for strings, however mypy wasn't able to infer that it is in fact a string.